src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/close_application_request.cc

```diff
@@ -0,0 +1,157 @@
+    app_mngr::ApplicationManager& application_manager,
+    app_mngr::rpc_service::RPCService& rpc_service,
+    app_mngr::HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handle)
+    : RequestFromHMI(message,
+                     application_manager,
+                     rpc_service,
+                     hmi_capabilities,
+                     policy_handle) {
+  subscribe_on_event(application_manager::event_engine::Event::EventID::
+                         BasicCommunication_CloseApplication,
+                     correlation_id());
+}
+
+CloseApplicationRequest::~CloseApplicationRequest() {}
+
+void CloseApplicationRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using namespace application_manager;
+
+  const uint32_t hmi_app_id = HMIAppId();
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(hmi_app_id);
+  if (!app) {
+    LOG4CXX_WARN(logger_,
+                 "Application with hmi_app_id " << hmi_app_id
+                                                << " isn't found.");
+    std::string info_message = std::string("Application with hmi_app_id ") +
+                               std::to_string(hmi_app_id) +
+                               std::string(" isn't found.");
+    SendErrorResponse(correlation_id(),
+                      static_cast<hmi_apis::FunctionID::eType>(function_id()),
+                      hmi_apis::Common_Result::GENERIC_ERROR,
+                      info_message);
+    return;
+  }
+
+  if (!app->IsRegistered()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Application with hmi_app_id " << hmi_app_id
+                                                 << " isn't registered.");
+    SendErrorResponse(correlation_id(),
+                      static_cast<hmi_apis::FunctionID::eType>(function_id()),
+                      hmi_apis::Common_Result::APPLICATION_NOT_REGISTERED,
+                      std::string("Application with hmi_app_id ") +
+                          std::to_string(hmi_app_id) +
+                          std::string(" isn't registered."));
+    return;
+  }
+
+  hmi_apis::Common_HMILevel::eType app_hmi_level =
+      static_cast<hmi_apis::Common_HMILevel::eType>(app->hmi_level());
+  if (hmi_apis::Common_HMILevel::NONE == app_hmi_level) {
+    LOG4CXX_DEBUG(logger_,
+                  "Application with hmi_app_id "
+                      << hmi_app_id << " is already has NONE hmi status.");
+    SendResponse(false,
+                 correlation_id(),
+                 static_cast<hmi_apis::FunctionID::eType>(function_id()),
+                 hmi_apis::Common_Result::IGNORED);
+    return;
+  }
+
+  app->set_hmi_level(mobile_apis::HMILevel::HMI_NONE);
+  LOG4CXX_DEBUG(logger_,
+                "Application with id " << app->app_id()
+                                       << " has changed hmi_level to NONE.");
+  MessageHelper::SendHMIStatusNotification(*app, application_manager_);
```

Comment:
@IGapchuk try to avoid using this function from message helper. Look into state_controller.